### ISSUE TITLE
[CI] TimeZoneRoundingTests#testRoundingRandom fails

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#27966]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+periodic/1149/console

```
   > Throwable #1: java.lang.AssertionError
   >    at __randomizedtesting.SeedInfo.seed([1E275DAB660CE5C0:B169E5B952C9109D]:0)
   >    at org.elasticsearch.common.rounding.TimeZoneRoundingTests.assertInterval(TimeZoneRoundingTests.java:589)
   >    at org.elasticsearch.common.rounding.TimeZoneRoundingTests.testRoundingRandom(TimeZoneRoundingTests.java:196)
```

This is a reproducible failure:
```
gradle :core:test \
  -Dtests.seed=1E275DAB660CE5C0 \
  -Dtests.class=org.elasticsearch.common.rounding.TimeZoneRoundingTests \
  -Dtests.method="testRoundingRandom" \
  -Dtests.security.manager=true \
  -Dtests.locale=et-EE \
  -Dtests.timezone=America/Grand_Turk
```

**Comment by DaveCTurner** [comment]
The offending time zone is `America/St_Johns` and the failing test is testing rounding in days - specifically rounding the chosen time point `1162091532554ms = 2006-10-29T03:12:12.554Z` up and down to the nearest day and checking that the chosen time is between the rounded times.

It isn't, however. It rounds down to `1162002600000ms = 2006-10-28T02:30:00Z` and rounds up to `1162089000000ms = 2006-10-29T02:30:00Z`, both of which are earlier than the chosen time point.

Of importance, surely, is that `America/St_Johns` changed their clocks at `2006-10-29T00:01` (local time) from UTC+02:30 to UTC+03:30. The rounded time points are midnight local time as expected.

**Comment by DaveCTurner** [comment]
Note to future self: because this time zone sets the clocks back at 00:01, it has the first minute of Sunday 2006-10-29 followed by 59 bonus minutes of Saturday 2006-10-28, and the chosen time is within that bonus period. The bug seems to be that we expect every time on a day to precede the following day's midnight within a given timezone, which is one of many reasonable-yet-false assumptions you can make about how time works.

**Comment by DaveCTurner** [comment]
We concluded in today's Search & Aggs meeting that the current behaviour is desirable and the assertion in the test is at fault. The fix is therefore to change the assertion.

As of TZDB 2017c, the only regular occurrences of these overlapping days since 1950 are the clocks going back in `America/St_Johns` (Newfoundland, 1987-2010), `America/Goose_Bay` (Labrador, 1987-2010) and `America/Moncton` (New Brunswick, 1993-2006) as described above. As a further bonus, in `America/St_Johns` in 1988 the clocks actually went back 2 hours at 00:01. Eastern Canada is a special place indeed.

There was also a one-off shift in Antarctica/Casey by three hours from `UTC+11` to `UTC+8` at `2010-03-05 02:00` local time which took it back to `2010-03-04` for an extra hour.

**Comment by DaveCTurner** [comment]
(also note that `Canada/Newfoundland` is an alias to `America/St_Johns` so suppress that too)

**Comment by DaveCTurner** [comment]
This will be fixed by #28151.
